farm/data_handler/processor.py

```diff
@@ -608,66 +608,98 @@ def file_to_dicts(self, file: str) -> [dict]:
-        text = dictionary["text"]
-        inputs = self.tokenizer.encode_plus(
-            text=text,
-            max_length=self.max_seq_len,
-            truncation=True,
-            add_special_tokens=True,
+    def dataset_from_dicts(self, dicts, indices=None, return_baskets=False, inference=False, debug=False):
+        # Tokenize in batches
+        texts = [x["text"] for x in dicts]
+        tokenized_batch = self.tokenizer.batch_encode_plus(
+            texts,
             return_offsets_mapping=True,
-            return_token_type_ids=True,
-            return_special_tokens_mask=True,
+            return_special_tokens_mask=True
         )
+        token_ids_batch = tokenized_batch["input_ids"]
+        segment_ids_batch = tokenized_batch["token_type_ids"]
 
-        # Get tokens as text with metadata
-        tokens = []
-        offsets = []
-        start_of_word = []
-        previous_token_end = -1
-        for token_id, is_special_token, offset in zip(inputs["input_ids"],
-                                                      inputs["special_tokens_mask"],
-                                                      inputs["offset_mapping"]):
-            if not is_special_token:
-                tokens.append(self.tokenizer.convert_ids_to_tokens(token_id))
-                offsets.append(offset[0])
-                start_of_word.append(True if offset[0] != previous_token_end else False)
-                previous_token_end = offset[1]
+        # From here we operate on a per sample basis
+        for dictionary, token_ids, segment_ids in zip(
+                dicts, token_ids_batch, segment_ids_batch
+        ):
 
-        token_dict = {"tokens": tokens,
-                      "offsets": offsets,
-                      "start_of_word": start_of_word}
+            # TODO Build tokenized dict for debug mode
+            tokenized = {}
 
-        if len(token_dict["tokens"]) == 0:
-            logger.warning(f"The following text could not be tokenized, likely because it contains a character that the tokenizer does not recognize: {text}")
-            return []
+            # Truncate results
+            token_ids = self.truncate(token_ids)
+            segment_ids = self.truncate(segment_ids)
 
-        # Build feature dict
-        input_ids, segment_ids = inputs["input_ids"], inputs["token_type_ids"]
+            # Pad results
+            input_ids, padding_mask, segment_ids = self.pad_sequences(token_ids, segment_ids)
+
+            feat_dict = {"input_ids": input_ids,
+                         "padding_mask": padding_mask,
+                         "segment_ids": segment_ids}
+
+            # Create labels
+            if not inference:
+                label_dict = self.generate_labels(dictionary)
+                feat_dict.update(label_dict)
+
+            self.add_basket(dictionary, tokenized, feat_dict)
 
+        if 0 in indices:
```

Comment:
Not sure why we use this condition here and if it really triggers the logging as we expect it

Reply:
Currently it is there because we still call `dataset_from_chunk`. With the conditional, we only print a single sample. Without it, we print for every chunk. If we get rid of dataset chunking, we will probably be able to remove this conditional

Reply:
Hmmmm it seems as though this condition has different implications for each task. Need to investigate further